splunk_eventgen/lib/plugins/output/httpevent.py

```diff
@@ -267,24 +298,19 @@ def flush(self, q):
                     payload.append(payloadFragment)
                 self.logger.debug("Finished processing events, sending all to splunk")
                 self._sendHTTPEvents(payload)
-                if self.config.httpeventWaitResponse:
-                    for session in self.active_sessions:
-                        response = session.result()
-                        if not response.raise_for_status():
-                            self.logger.debug("Payload successfully sent to httpevent server.")
-                        else:
-                            self.logger.error("Server returned an error while trying to send, response code: %s" %
-                                              response.status_code)
-                            raise BadConnection(
-                                "Server returned an error while sending, response code: %s" % response.status_code)
-                else:
+                if not self.config.httpeventWaitResponse:
                     self.logger.debug("Ignoring response from HTTP server, leaving httpevent outputter")
+                else:
+                    for session_info in self.active_session_info:
+                        url, session = session_info[0], session_info[1]
+                        try:
+                            response = session.result(3)
+                            self.reset_count(url)
```

Comment:
if reset the count each time for one specific url, the url seems will not be removed forever?

Reply:
I want to prevent a working server to be dropped because of some flakiness. I have seen that Splunk (without an ample size queue) returns error responses even though it is alive. Therefore, even after a series of failures, if we see it working again, we don't remove it.

Reply:
make sense